python/cudf/cudf/core/multiindex.py

```diff
@@ -483,15 +483,11 @@ def copy(
         else:
             names = self.names
         if self._levels is not None:
-            levels: None | list[cudf.Index] = [
-                idx.copy(deep=deep) for idx in self._levels
-            ]
+            levels = [idx.copy(deep=deep) for idx in self._levels]
```

Comment:
In theory, we can remove this if/else block here and just always use this. Likewise for codes down below.

But this was getting a bit far afield from fixing this specific issue and I didn't want to change too much.